web/api/v1/api.go

```diff
@@ -147,6 +148,8 @@ func (api *API) Register(r *route.Router) {
 
 	r.Get("/targets", instr("targets", api.targets))
 	r.Get("/alertmanagers", instr("alertmanagers", api.alertmanagers))
+
+	r.Get("/read", prometheus.InstrumentHandler("read", http.HandlerFunc(api.remoteRead)))
```

Comment:
Yeah, all the remote stuff is still experimental and I am expecting/hoping for at least one breaking change in the 2.x timeframe.